### PR TITLE
Fixes #34260 - add rex features dropdown to host page

### PR DESCRIPTION
**Pull request by amirfefer**
https://user-images.githubusercontent.com/11807069/149392178-43bb0abf-42da-4742-98e9-1a70c385bde3.mov

### CODE REVIEW

File: webpack/react_app/components/FeaturesDropdown/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+const FeaturesDropdown = ({ hostId }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const {
+    response: { results: features },
+    status,
+  } = useAPI('get', foremanUrl(REX_FEATURES_API));
+  const dispatch = useDispatch();
+  const dropdownItems = features
+    ?.filter(feature => feature.host_action_button)
```
Comment (amirfefer) [inline webpack/react_app/components/FeaturesDropdown/index.js:28]: @adamruzicka I tried to filter this by the API call, with no success
I added `scoped_search :on => :host_action_button` 
and created an API call to `/api/remote_execution_features?search=host_action_button=true` 

We can filter it on the client, but if the server can return just the relevant features, it just better :)

File: webpack/react_app/components/FeaturesDropdown/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  ];
+
+  return (
+    <Dropdown
+      alignments={{ default: 'right' }}
+      onSelect={() => setIsOpen(false)}
+      toggle={
+        <DropdownToggle
+          splitButtonItems={scheduleJob}
+          toggleVariant="secondary"
```
Comment (amirfefer) [inline webpack/react_app/components/FeaturesDropdown/index.js:54]: looks like it doesn't work... no `secondary` nor `primary`

Comment (jeremylenz) [reply webpack/react_app/components/FeaturesDropdown/index.js:54]: I've noticed this too. Maybe we should file a PF issue..

Comment (jeremylenz) [reply webpack/react_app/components/FeaturesDropdown/index.js:54]: I made [this Codepen](https://codesandbox.io/s/holy-breeze-70rwf?file=/index.js) intending to file an issue, but `toggleVariant` seems to work fine there...

File: webpack/react_app/components/FeaturesDropdown/index.js (hunk 1 above)
Comment (adamruzicka) [reply webpack/react_app/components/FeaturesDropdown/index.js:28]: If I'm reading things right the controller currently does not support searching, tracking it as[1]. We could implement that, but for now let's go with frontend only filtering. Standard Satellite deployment has around 20 features, so nothing terrible.

[1] - https://projects.theforeman.org/issues/34284

File: app/models/remote_execution_feature.rb [hunk 3]
```diff
@@ -10,6 +10,7 @@ class RemoteExecutionFeature < ApplicationRecord
   friendly_id :label
 
   scope :with_host_action_button, -> { where(:host_action_button => true) }
+  scoped_search :on => :host_action_button
```
Comment (ezr-ondrej) [inline app/models/remote_execution_feature.rb:13]: As of @adamruzicka's answer, I'd keep this out of this PR, as it feels out of scope, especially when not used :)

File: webpack/react_app/components/FeaturesDropdown/index.js (hunk 2 above)
Comment (amirfefer) [reply webpack/react_app/components/FeaturesDropdown/index.js:54]: So...the `toggleVariant` was added in `@patternfly/react-core@4.177.0`
currently we use `"@patternfly/react-core": "~4.162.2"`
:\

anyway, it's not a bug and should be resolved as soon as we will update pf

File: app/models/remote_execution_feature.rb (hunk 3 above)
Comment (amirfefer) [reply app/models/remote_execution_feature.rb]: removed :)

File: webpack/react_app/components/FeaturesDropdown/index.js (hunk 2 above)
Comment (jeremylenz) [reply webpack/react_app/components/FeaturesDropdown/index.js:54]: But the old prop `isPrimary` doesn't seem to do anything either :thinking:

**Comment by ezr-ondrej** [comment]
So shall we bump patternfly and package before we get this in?

**Comment by jeremylenz** [comment]
> So shall we bump patternfly and package before we get this in?
I'd be +1 to that :+1:

**Comment by ares** [comment]
Any progress on this? Who has the ball for patternfly bump?

**Comment by amirfefer** [comment]
upgrading all PF4 related libraries - https://github.com/theforeman/foreman-js/pull/378

**Comment by amirfefer** [comment]
Worlks well with https://github.com/theforeman/foreman/pull/9110
Screen Shot 2022-02-21 at 15 14 07

**Comment by amirfefer** [comment]
odd, rubocop is passed locally 🤔

**Comment by adamruzicka** [comment]
> odd, rubocop is passed locally thinking

That might have been me with this https://github.com/theforeman/foreman_remote_execution/commit/4d4957a901741e2fe6e7646b0b8027fc8b01006d , https://github.com/theforeman/foreman_remote_execution/pull/695 should fix it.

**Comment by adamruzicka** [comment]
Could you rebase plase? It should make the rubocop issue go away

**Comment by amirfefer** [comment]
@adamruzicka - rebased, but still, https://github.com/theforeman/foreman/pull/9110 hasn't merged yet.

**Comment by ofedoren** [comment]
It is merged now :rocket:

**Review by ezr-ondrej — APPROVED** [review]
Thanks @amirfefer for doing this and thaks all the reviewers for the great coordinated effort to get this in properly! :+1:

File: webpack/react_app/components/FeaturesDropdown/actions.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { foremanUrl } from 'foremanReact/common/helpers';
+import { sprintf, translate as __ } from 'foremanReact/common/I18n';
+import { post } from 'foremanReact/redux/API';
+
+export const runFeature = (hostId, feature, label) => dispatch => {
+  const url = foremanUrl(
+    `/job_invocations?feature=${feature}&host_ids%5B%5D=${hostId}`
```
Comment (ezr-ondrej) [inline webpack/react_app/components/FeaturesDropdown/actions.js:7]: Unfortunatelly this is wrong as we are passing the Host name no an Id here.
That's something I've complained a lot. In core we are passing the host identifier as hostId prop all around the page, but it is usually a name, not an ID, I believe it should be fixed to properly denote what we are actually passing to this prop. @amirfefer could you fix this instance, so it would prefil the search for the job correctly?
@jeremylenz ^ for awarnes about the mess and what it can cause ;)

Comment (jeremylenz) [reply webpack/react_app/components/FeaturesDropdown/actions.js:7]: @ezr-ondrej, Yeah, I noticed quite a while back that `host_ids` is named poorly and is actually just a scoped search.  That's the way we treat it in Katello:

```js
`host_ids=name ^ (${hostname})`
```

https://github.com/Katello/katello/blob/2f9e60b313c9cd05f44811db2939ed66fc61e0ac/webpack/components/extensions/HostDetails/Tabs/customizedRexUrlHelpers.js#L13-L14

Let me know if you're going to break this so I can make sure it gets fixed ;)

Comment (ezr-ondrej) [reply webpack/react_app/components/FeaturesDropdown/actions.js:7]: it is not enrirely true, it works now when we made sure, that hostId is a numerical Id :)
